xlators/features/shard/src/shard.c

```diff
@@ -7329,8 +7329,8 @@ shard_unlink_handler_spawn(xlator_t *this)
         if (ti->running) {
             pthread_cond_signal(&ti->cond);
         } else {
-            ret = gf_thread_create(&ti->thread, NULL, shard_unlink_handler, ti,
-                                   "shard_unlink");
+            ret = gf_thread_create(&ti->thread, NULL, shard_unlink_handler,
+                                   priv, "shard_unlink");
```

Comment:
IMO, you can still pass `ti `and access `priv->deletion_interval `from` ti->this`

Reply:
OK